core/src/commonMain/kotlin/com/revenuecat/purchases/kmp/models/SubscriptionOption.kt

```diff
@@ -66,7 +66,7 @@ public val SubscriptionOption.billingPeriod: Period?
  * Not applicable for Amazon subscriptions.
  */
 public val SubscriptionOption.isPrepaid: Boolean
-    get() = fullPricePhase?.recurrenceMode == RecurrenceMode.NON_RECURRING
+    get() = fullPricePhase?.recurrenceMode?.name == RecurrenceMode.NON_RECURRING.name
```

Comment:
This is actually something I wasn't fully understanding... I was getting a compile time error when comparing the enum values themselves because it was different types of enums... However, the IDE didn't mark this as an error and as I navigated, everything seem to be the same... This should work for now until we actually move `SubscriptionOption` to not use `expect`/`actual` which should avoid this, but if you know what might be happening, it would be great 🙏

Reply:
Yea I know what's happening haha:
* The `recurrenceMode` extension is shadowed by the existing member field of the same name.
* The `recurrenceMode` extension is a different type than the existing member field.
* When compiling this for Android `PricingPhase.recurrenceMode` refers to the existing member field on Android, and thus its type is `com.revenuecat.purchases.models.RecurrenceMode`.
* The IDE sees the extension, and the shadowing warning is suppressed (`EXTENSION_SHADOWED_BY_MEMBER`), so the IDE shows `recurrenceMode` as if its type is `com.revenuecat.purchases.kmp.models.RecurrenceMode` (notice the `kmp` package).

These were some of the earliest reported issues in KobanKat haha: 
* #48 
* #51 

I created [SDK-3464](https://linear.app/revenuecat/issue/SDK-3464/detekt-rule-to-error-when-shadows-change-types) to _detekt_ these issues, but if we move away from `expect`/`actual` I _expect_ that this is not needed anymore as we avoid these types of issues.

Reply:
Ohhh makes sense thanks for explaining! And yeah, I think when we move away from `expect`/`actual` this sholdn't be a problem, so maybe it's ok to leave as is for now then...

Reply:
Yes I think it's fine to leave it as-is, as long as we don't release it in this state haha